src/visualizations/config/adapters/dhis_highcharts/customSVGOptions/singleValue/renderer/addIconElement.js

```diff
@@ -0,0 +1,64 @@
+const parser = new DOMParser()
+
+export function addIconElement(svgString, color) {
+    const svgIconDocument = parser.parseFromString(svgString, 'image/svg+xml')
+    const iconElHeight = svgIconDocument.documentElement.getAttribute('height')
+    const iconElWidth = svgIconDocument.documentElement.getAttribute('width')
+    const iconGroup = this.renderer
+        .g('icon')
+        .attr({ color, 'data-test': 'visualization-icon' })
+        .css({
+            visibility: 'hidden',
+        })
+
+    /* Force the group element to have the same dimensions as the original
+     * SVG image by adding this rect. This ensures the icon has the intended
+     * whitespace around it and makes scaling and translating easier. */
+    this.renderer.rect(0, 0, iconElWidth, iconElHeight).add(iconGroup)
+
+    /* The code below makes some assumptions about icon SVGs:
+     * 1. The SVG children are all <path> elements and they are not nested
+     * 2. Only the `d`, `fill-rule`, `clip-rule` and `fill` attributes are used */
+    Array.from(svgIconDocument.documentElement.children).forEach((pathNode) => {
+        /* Highcharts expects an array of letters and numbers but the
+         * d-attribute string only puts spaces in between adjacent number,
+         * not between letter and numbers. */
+        const d = pathNode
+            .getAttribute('d')
+            // Add space after letter when followed by number
+            .replace(/([A-Z])([0-9])/g, '$1 $2')
```

Comment:
Yes. We discussed this part during the review call. We decided to just the simpler native DOM implementation instead of this error-prone approach. See [489ea08](https://github.com/dhis2/analytics/pull/1698/commits/489ea087c766b7e0ac736359373d4b11033a3548)